### PR TITLE
Add python api for CTCGreedyDecoder

### PR DESCRIPTION
**Pull request by iimironov**
Ticket 43149

### CODE REVIEW

File: ngraph/python/src/ngraph/opset6/ops.py [hunk 1]
```diff
@@ -58,6 +58,35 @@
 # -------------------------------------------- ops ------------------------------------------------
 
 
+@nameable_op
+def ctc_greedy_decoder_seq_len(
+        data: NodeInput,
+        sequence_length: NodeInput,
+        blank_index: Optional[int] = 0,
```
Comment (jiwaszki) [inline ngraph/python/src/ngraph/opset6/ops.py:65]: As far as I understand default value for `blank_index` is equal to `num_classes - 1`. Is it set to this value while processing in cpp internals? If not then I think you should change 0 to `None`, put an if statement to get length of the 3rd dimension from `data` and subtract one.

File: ngraph/python/src/ngraph/opset6/ops.py [hunk 2]
```diff
@@ -58,6 +58,35 @@
...
+        merge_repeated: bool = True,
+        classes_index_type: str = "i32",
+        sequence_length_type: str = "i32",
+        name: Optional[str] = None,
+) -> Node:
+    """Return a node which performs CTCGreedyDecoderSeqLen.
+
+    @param data:            The input tensor. Shape: [batch_size, seq_length, num_classes]
```
Comment (jiwaszki) [inline ngraph/python/src/ngraph/opset6/ops.py:73]: Optional: add 3D before tensor

File: ngraph/python/src/ngraph/opset6/ops.py [hunk 3]
```diff
@@ -58,6 +58,35 @@
...
+    @param sequence_length: Input tensor with sequence length. Shape: [batch_size]
```
Comment (jiwaszki) [inline ngraph/python/src/ngraph/opset6/ops.py:74]: Optional: add 1D before tensor

**Review by jiwaszki — CHANGES_REQUESTED** [review]

File: ngraph/python/src/ngraph/opset6/ops.py [hunk 4]
```diff
@@ -58,6 +58,35 @@
 # -------------------------------------------- ops ------------------------------------------------
 
 
+@nameable_op
+def ctc_greedy_decoder_seq_len(
+        data: NodeInput,
+        sequence_length: NodeInput,
+        blank_index: Optional[int] = None,
```
Comment (jiwaszki) [inline ngraph/python/src/ngraph/opset6/ops.py:65]: This is fine.

File: ngraph/python/tests/test_ngraph/test_create_op.py [hunk 5]
```diff
@@ -77,6 +77,36 @@ def test_ctc_greedy_decoder(dtype):
...
+                             (np.float32, np.int32, "i32", "i32", True),
+                             (np.float32, np.int32, "i64", "i32", True),
+                             (np.float32, np.int32, "i32", "i64", True),
+                             (np.float32, np.int32, "i64", "i64", True),
+                             (np.float64, np.int64, "i32", "i32", False),
+                             (np.float64, np.int64, "i64", "i32", False),
+                             (np.float64, np.int64, "i32", "i64", False),
+                             (np.float64, np.int64, "i64", "i64", False)
+                         ],)
+def test_ctc_greedy_decoder_seq_len(fp_dtype, int_dtype, int_ci, int_sl, merge_repeated):
```
Comment (lazarevevgeny) [inline ngraph/python/tests/test_ngraph/test_create_op.py:91]: Please, add tests for case when `blank_index` is `None`

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: ngraph/python/src/ngraph/opset6/ops.py [hunk 6]
```diff
@@ -58,6 +58,35 @@
...
+                            Optional parameter. Default value is num_classes-1.
+    @return:                The new node which performs CTCGreedyDecoderSeqLen.
+    """
+    inputs = as_nodes(data, sequence_length, blank_index)
+
+    attributes = {
+        "merge_repeated": merge_repeated,
+        "classes_index_type": classes_index_type,
+        "sequence_length_type": sequence_length_type
+    }
```
Comment (jiwaszki) [inline ngraph/python/src/ngraph/opset6/ops.py:85]: I assume all tests for `blank_index = False` are failing due to recognizing `None` value as `dtype('O')`. You should add some check before setting inputs, please add `if blank_index is None` and assign default value of `num_classes-1` to it if true.

**Review by jiwaszki — CHANGES_REQUESTED** [review]

File: ngraph/python/tests/test_ngraph/test_create_op.py [hunk 7]
```diff
@@ -77,6 +77,46 @@ def test_ctc_greedy_decoder(dtype):
...
+    input0_shape = [8, 20, 128]
+    input1_shape = [8]
+    input2_shape = [1]
+    expected_shape = [8, 20]
+
+    parameter_input0 = ng.parameter(input0_shape, name="Input0", dtype=fp_dtype)
+    parameter_input1 = ng.parameter(input1_shape, name="Input1", dtype=int_dtype)
+    parameter_input2 = ng.parameter(input2_shape, name="Input2", dtype=int_dtype)
+    if blank_index is False:
+        parameter_input2 = input0_shape[2] - 1
```
Comment (jiwaszki) [inline ngraph/python/tests/test_ngraph/test_create_op.py:109]: You get the idea right. However this should be implemented in _ngraph/python/src/ngraph/opset6/ops.py_ inside _ctc_greedy_decoder_seq_len_ function. Right now all we have is a test mock that does not work across nGraph module thus outside of this this test `None` value will still throw errors.

**Review by jiwaszki — CHANGES_REQUESTED** [review]

**Review by jiwaszki — APPROVED** [review]
Looks fine! Just wait for green checks and it is ready to merge.

**Review by lazarevevgeny — APPROVED** [review]

**Review by pavel-esir — APPROVED** [review]
